spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/core/DynamicTopicDestinationResolver.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.sns.core;
+
+import io.awspring.cloud.core.resource.AmazonResourceName;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.CreateTopicRequest;
+import software.amazon.awssdk.services.sns.model.ListTopicsRequest;
+import software.amazon.awssdk.services.sns.model.ListTopicsResponse;
+import software.amazon.awssdk.services.sns.model.Topic;
+
+import org.springframework.messaging.core.DestinationResolutionException;
+import org.springframework.messaging.core.DestinationResolver;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Agim Emruli
+ * @author Alain Sahli
+ * @since 1.0
+ */
+public class DynamicTopicDestinationResolver implements DestinationResolver<String> {
+
+	private final SnsClient snsClient;
+
+	private boolean autoCreate;
+
+	public DynamicTopicDestinationResolver(SnsClient snsClient) {
+		this.snsClient = snsClient;
+	}
+
+	public void setAutoCreate(boolean autoCreate) {
```

Comment:
`autoCreate` is never set. Should we expose it in `SnsProperties`?

Reply:
I think so and by default the property should be true?

Reply:
I vote that we put it default true. Since names can be resolved this way as well. Currently, I made it that if the string starts with arn it will skip topic creation since it will cause an exception. If the same topic name is tried to be created it will return arn of the topic so in sense we can resolve convertAndSend by name as well. It is not the optimal way tho. Now thinking mb it is better to check if the destination starts with arn: